tests/integration_tests/modules/test_cli.py

```diff
@@ -45,10 +45,7 @@ def test_valid_userdata(client: IntegrationInstance):
     assert result.ok
     assert "Valid schema user-data" in result.stdout.strip()
     result = client.execute("cloud-init status --long")
-    # Ubuntu Noble will exit 2 for status due to cloud-init's warning about
-    # disabling /etc/apt/sources.list in favor of deb822 sources
-    return_code = 2 if CURRENT_RELEASE.series == "noble" else 0
-    assert return_code == result.return_code, (
+    assert 0 == result.return_code, (
```

Comment:
This still won't actually be true until cloudimages are updated in noble because https://code.launchpad.net/~juliank/livecd-rootfs/+git/livecd-rootfs/+merge/458366 just merged yesterday. But, I agree this is this desired end-goal. And we might as well set those expectations now and expect that this integration test will go green on each cloud platform as those cloudimage streams publish new Ubuntu Noble images.


Today, what we will still see is failures against daily images because /etc/apt/sources.list is still being removed in images by cloud-init. That will change once images with /etc/cloud/build.info  >= `version: 20240213` are present and https://github.com/canonical/cloud-init/pull/4770 is published to Noble.

But I expect for the moment that this integration test will still fail:
```
CLOUD_INIT_OS_IMAGE=noble tox -e integration-tests -- tests/integration_tests/modules/test_cli.py::test_valid_userdata
```

Note that installing from the daily PPA will not fail because as part of that process we install a new cloud-init deb and invoke cloud-init clean --logs. So you'd see it passing today with:
```
CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/daily CLOUD_INIT_OS_IMAGE=noble tox -e integration-tests -- tests/integration_tests/modules/test_cli.py::test_valid_userdata
```
```